activerecord/lib/active_record/connection_adapters/mysql/database_statements.rb

```diff
@@ -17,13 +17,10 @@ def select_all(arel, name = nil, binds = [], preparable: nil)
         # as values.
         def select_one(arel, name = nil, binds = [])
           arel, binds = binds_from_relation(arel, binds)
-          @connection.query_options.merge!(as: :hash)
```

Comment:
This is necessary. See https://github.com/rails/rails/pull/23461#discussion_r60340904